tests/rptest/tests/rbac_test.py

```diff
@@ -236,6 +237,41 @@ def get_role_succeeds():
 
         # TODO: add tests for role membership once assigning members to roles gets implemented
 
+    @cluster(num_nodes=3)
+    def test_update_role(self):
+        with expect_role_error(RoleErrorCode.ROLE_NOT_FOUND):
+            self.superuser_admin.update_role(
+                role=self.role_name0, update=RoleUpdate(role=self.role_name1))
+
+        self._create_and_wait_for_role(role=self.role_name0)
+
+        res = self.superuser_admin.update_role(
+            role=self.role_name0, update=RoleUpdate(role=self.role_name1))
+        new_role = res.json()['role']
+        assert new_role == self.role_name1, f"Unexpected role name: {new_role} != {self.role_name1}"
+
+        def update_completes_atomically():
+            # Fail if we ever see both roles at the same time
+            assert self._set_of_user_roles() in [{self.role_name0},
+                                                 {self.role_name1}]
```

Comment:
This check seems redundant give the next check for set equivalence.

Reply:
This is to guard against seeing the intermediary state of both roles being present. I've updated the comment to make this even clearer.